src/components/common/CodeReview.js

```diff
@@ -155,20 +239,32 @@ class CodeReview extends Component {
   }
 
   renderCodeReview(code, commentList) {
+    const { isFeedbackView, isTeacherView, botComments, teacherComments } =
+      this.props;
+    const { focusedId } = this.state;
     const highlightedCode = CodeReview.highlightCode(code, 'python').split(
       '\n',
     );
     return highlightedCode.map((line, i) => {
-      const lineComments = commentList.filter(
-        (comment) => comment.data.line === i + 1,
+      const filteredComments = commentList.filter(
+        (comment) =>
+          (isTeacherView && comment._id === focusedId) || !isTeacherView,
       );
+      const lineComments = [
+        ...teacherComments.filter((comment) => comment.data.line === i + 1),
+        ...botComments.filter((comment) => comment.data.line === i + 1),
+        ...filteredComments.filter((comment) => comment.data.line === i + 1),
+      ];
+
       const renderedComments = this.renderCommentList(lineComments);
       return (
+        // add a key for each line ...
```

Comment:
Is this a todo? You can add a simple key, should be fine.

Reply:
Yes i tried, but the problem is that it does not accept using the index of the line as a key since it can cause some inconsistent re-renderings if a line is delete for example... I thaught about using a hash of the line including the line index since it sould be different for every line, but that seemed more complex than necessary ... 

So yes kind of a "todo but not sure how" situation ...